app/src/main/java/com/github/sdp_begreen/begreen/models/event/Event.kt

```diff
@@ -1,13 +1,13 @@
 package com.github.sdp_begreen.begreen.models.event
 
 import com.github.sdp_begreen.begreen.firebase.RootPath
-import com.github.sdp_begreen.begreen.models.CustomLatLng
 import com.github.sdp_begreen.begreen.firebase.utils.CopyableWithId
+import com.github.sdp_begreen.begreen.models.CustomLatLng
 
 /**
  * Interface that define the base attribute that an Event should have
  */
-interface Event<T>: CopyableWithId<T> {
+interface Event<T> : CopyableWithId<T> {
```

Comment:
The `Event` class doesn't have the `id` field because it implements `CopyableWithId` but `EventParticipant` does have the `id` field and doesn't implement `CopyableWithId`. I think that the difference comes from whether an object should be copiable with a new `id` or not, but why don't we make them all copiable ? I am not a big fan of this `CopyableWithId` if we use it in some cases and not in others since it introduces some confusion, but anyway it's not a big deal.

Reply:
I defined this `CopyableWithId` interface, because in the [`EventServiceImpl`](https://github.com/SDP-BeGreen/BeGreen/blob/e984da5b556774e6669d1ba65982780f112ba3fd/app/src/main/java/com/github/sdp_begreen/begreen/firebase/eventServices/EventServiceImpl.kt#L28), in the `createEvent` method, instead of mutating the id field, I prefer to copy the `Event` with a new id. All `data class` have a default copy method, but I needed a way to tell that an `Event` should also be copyable with a new id. That's why I added this `CopyableWithId` interface.

As I work a bit differently with the `EventParticipant`, and I never expressly copy them with a new Id, I didn't see the purpose of making `EventParticipant` implement the `CopyableWithId` interface, to offer a method that would never be called.

Let me know if you have an idea to maybe make it more clear ?

Reply:
No you can keep it like that it's fine. But we could think about changing other data classes of the project that need this copy. For example, in `FirebaseDB` in the method `addTrashPhoto` we have the following line :

`var newPhotoMetadata = trashPhotoMetadata.copy(pictureId = null)`

We can make `TrashPhotoMetadata` implement `CopyableWithId`, and go through the project to see if there are other classes that should implement this interface. 

For now, it doesn't concern your task so you don't have to change it, but we should envisage to have a small task in the final sprint to clean up some parts of the project (only small things that doesn't change the architecture or break something)